policy-test/src/curl.rs

```diff
@@ -252,6 +250,38 @@ impl Running {
         code
     }
 
+    /// Waits for the curl container to complete and returns the http status
+    /// code.
+    pub async fn http_status_code(self) -> String {
+        self.inits_complete().await;
+        let api = kube::Api::namespaced(self.client.clone(), &self.namespace);
+
+        let pod = self.finished(&api).await;
+        let status_code = api
+            .logs(
+                &pod.name_unchecked(),
+                &LogParams {
+                    container: Some("curl".to_string()),
+                    ..Default::default()
+                },
+            )
+            .await
+            .expect("must be able to get curl logs")
+            .lines()
+            .last()
+            .expect("curl logs are empty")
+            .to_owned();
```

Comment:
we could even try turning this into a proper status code with [`http::StatusCode::try_from`](https://docs.rs/http/0.1.1/http/status/struct.StatusCode.html#impl-HttpTryFrom%3C&'a%20str%3E)